drivers/adc/adc_demo/iio_adc_demo.h

```diff
@@ -42,10 +42,29 @@
 
 #include <stdlib.h>
 #include "adc_demo.h"
+#include "iio.h"
 
-struct iio_device adc_demo_iio_descriptor = {
-	.debug_reg_read = (int32_t (*)()) adc_demo_reg_read,
-	.debug_reg_write = (int32_t (*)()) adc_demo_reg_write
+#define MAX_NR_CHANNELS 10
+
+#define ADC_DEMO_DEV(_numch) {\
+	.num_ch = _numch, \
+	.attributes = NULL,	\
+	.debug_attributes = NULL,	\
+	.buffer_attributes = NULL,	\
+	.prepare_transfer = init_adc_channels,	\
```

Comment:
update_active_adc_channels should go here

Reply:
the number of channels is now configurable from the init param, and that function retrieves the actual number of active channels

Reply:
prepare_transfer shuld only update the mask with the activ channels, it is how iio works.

When selecting from the oscilloscope channel 0 the mask will be 0x00000001,  when selecting channel 1, it will be 0x00000002 and if both will be 0x00000003

Reply:
My bad, I found this out and I also had to modify some other things and now it has update_adc_channels